File: PersonalizedAndroid/src/com/example/personalizedandroid/WelcomeActivity.java
package com.example.personalizedandroid;

import android.content.Intent;
import android.os.Bundle;
import android.os.UserManager;
import android.util.Log;
import android.widget.Button;
import android.widget.TextView;
import android.app.Activity;
import android.content.pm.UserInfo;

public class WelcomeActivity extends Activity {
    private static final String TAG = "PersonalizedAndroid";
    private UserManager mUserManager;
    public static final String EXTRA_USER_ID = "extra_user_id";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        TextView welcomeText = findViewById(R.id.welcome_text);
        Button continueButton = findViewById(R.id.continue_button);

        // Initialize UserManager
        mUserManager = getSystemService(UserManager.class);

        // Set welcome message with current user's name
        UserInfo currentUser = getCurrentUser();
        if (currentUser != null) {
            Log.d(TAG, "WelcomeActivity: Current user ID: " + currentUser.id + ", Name: " + currentUser.name);
            welcomeText.setText("Welcome, " + currentUser.name + "!");
        } else {
            Log.e(TAG, "WelcomeActivity: No current user found");
            welcomeText.setText("Welcome!");
        }

        // Navigate to MainActivity on button click
        continueButton.setOnClickListener(v -> {
            Intent intent = new Intent(WelcomeActivity.this, MainActivity.class);
            if (currentUser != null) {
                intent.putExtra(EXTRA_USER_ID, currentUser.id);
            }
            startActivity(intent);
            finish();
        });
    }

    private UserInfo getCurrentUser() {
        if (mUserManager != null) {
            UserInfo userInfo = mUserManager.getUserInfo(android.os.Process.myUserHandle().getIdentifier());
            if (userInfo != null) {
                Log.d(TAG, "getCurrentUser: ID: " + userInfo.id + ", Name: " + userInfo.name);
            } else {
                Log.e(TAG, "getCurrentUser: UserInfo is null");
            }
            return userInfo;
        }
        Log.e(TAG, "getCurrentUser: UserManager is null");
        return null;
    }
}

File: PersonalizedAndroid/PersonalizedAndroid/src/com/example/personalizedandroid/WelcomeActivity.kt
package com.example.personalizedandroid

import android.app.Activity
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.os.UserManager
import android.util.Log
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import android.content.pm.UserInfo

class WelcomeActivity : Activity() {
    private lateinit var userManager: UserManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)

        val welcomeLayout: LinearLayout = findViewById<LinearLayout>(R.id.welcome_layout)!!
val welcomeText: TextView = findViewById<TextView>(R.id.welcome_text)!!
val continueButton: Button = findViewById<Button>(R.id.continue_button)!!


userManager = getSystemService(UserManager::class.java)!!

        val currentUser = getCurrentUser()
        if (currentUser != null) {
            Log.d(TAG, "WelcomeActivity: Current user ID: ${currentUser.id}, Name: ${currentUser.name}")
            welcomeText.text = "Welcome, ${currentUser.name}!"
            applyTheme(currentUser.id, welcomeLayout)
        } else {
            Log.e(TAG, "WelcomeActivity: No current user found")
            welcomeText.text = "Welcome!"
            welcomeLayout.setBackgroundResource(R.drawable.gradient_blue)
        }

        continueButton.setOnClickListener {
            val intent = Intent(this, MainActivity::class.java)
            currentUser?.let {
                intent.putExtra(EXTRA_USER_ID, it.id)
            }
            startActivity(intent)
            finish()
        }
    }

    private fun getCurrentUser(): UserInfo? {
        return userManager.getUserInfo(android.os.Process.myUserHandle().identifier)?.also {
            Log.d(TAG, "getCurrentUser: ID: ${it.id}, Name: ${it.name}")
        } ?: run {
            Log.e(TAG, "getCurrentUser: UserInfo is null")
            null
        }
    }

    private fun applyTheme(userId: Int, layout: LinearLayout) {
        when (loadUserPreferences(userId)) {
            "solid_red" -> layout.setBackgroundResource(R.drawable.solid_red)
            "solid_green" -> layout.setBackgroundResource(R.drawable.solid_green)
            else -> layout.setBackgroundResource(R.drawable.gradient_blue)
        }
    }

    private fun loadUserPreferences(userId: Int): String {
        val prefs: SharedPreferences = getSharedPreferences("user_$userId", MODE_PRIVATE)
        return prefs.getString("theme", "gradient_blue") ?: "gradient_blue"
    }

    companion object {
        private const val TAG = "PersonalizedAndroid"
        const val EXTRA_USER_ID = "extra_user_id"
    }
}

